internal/utils/file_write_data_utils.go

```diff
@@ -6,15 +6,17 @@ import (
 )
 
 /*
-Directory where write buffer data will be saved to disk before uploaded to a cloud bucket.
-This directory needs to be shared across files so all functions that access it will be defined here.
+Subfolder where write buffer data will be saved to disk before uploaded to a cloud bucket.
+This subfolder needs to be shared across files so all functions that access it will be defined here.
 */
-const temp_write_buf_dir = "tmp/temp_write_buffers"
+
+const write_buffer_folder = "temp_write_buffers"
```

Comment:
Perhaps, rather than being a global variable, the functions on this file should be methods on an object which stores the write buffer folder and gets passed around. This would also allow using custom temp dirs during testing.

Reply:
I tried not to pass another object in since the object would have to be passed all the way down (or up from) the strace level. I think it should be okay not to allow having a custom temp dirs sincec these files will be delted before the program exits anyways?